src/core/data-structures/signals.js

```diff
@@ -15,6 +15,7 @@
  */
 
 import {Deferred} from './promise';
+import {Timestamp} from '../types/date';
```

Comment:
Type annotations could instead use `../types/date.Timestamp` for less-frequently used typedefs, but this felt common enough to merit the import